app/main.py

```diff
@@ -1 +1,47 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+    alive: list[Animal] = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self._health = health
+        self.name = name
+        self.hidden = hidden
+        if self not in Animal.alive:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    def die(self) -> None:
```

Comment:
don`t think, that this method is necessary, you can just delete Animal from list in method Bite. But if AI ok with this, then me too